src/main.rs

```diff
@@ -183,20 +201,31 @@ fn run() -> Result<(), Error> {
     // initialize the transaction broadcast channel
     let (broadcast_sender, broadcast_receiver) = futures::sync::mpsc::unbounded();
 
+    let shared_stats = SharedStats::default();
+
     let transaction_task = {
         let tpool = tpool.clone();
         let blockchain = blockchain.clone();
+        let shared_stats = shared_stats.clone();
         tasks.task_create_with_inputs("transaction", move |r| {
-            transaction_task(blockchain, tpool, r)
+            transaction_task(blockchain, tpool, r, shared_stats)
         })
     };
 
     let block_task = {
         let blockchain = blockchain.clone();
         let clock = clock.clone();
         let selection = Arc::clone(&selection);
+        let shared_stats = shared_stats.clone();
         tasks.task_create_with_inputs("block", move |r| {
-            block_task(blockchain, selection, clock, r, broadcast_sender)
+            block_task(
+                blockchain,
+                selection,
+                clock,
+                r,
+                broadcast_sender,
+                shared_stats,
+            )
```

Comment:
It may improve maintainablility to group some or all of these these parameters into a struct at this point.

Reply:
I agree, 5 parameters are hard to follow. Should I refactor it now or wait with it until the demo? We will have to do quite some clean up then and developing fresh code is slightly easier with unpacked args.

Reply:
Whichever works for you. I'd write even new code with easy to read structures that can be autocompleted.